packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2423,6 +2423,15 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       final Size size = renderEditable.size;
       final Matrix4 transform = renderEditable.getTransformTo(null);
       _textInputConnection!.setEditableSizeAndTransform(size, transform);
+      final TextSpan textSpan = buildTextSpan();
```

Comment:
This is to get the length of the text for use on line 2430

Reply:
Oh I see to implement `closestPositionToPoint` and friends you will need the bounding boxes for each glyph. This is going to be pretty heavy for long documents, and since a `RenderEditable` clips its content if it gets too long (i.e. `RenderEditable` is a scrollable container and only part of it will remain visible), you'll get false positives on glyphs that are not currently visible.

Reply:
What scribble function(s) would it break if we leave the `closestPositionToPoint` family unimplemented?

Reply:
Oh interesting. I will have to put some long text in and see how that affects things.

So, `closestPositionToPoint` itself allows UIKit to move the selection around based on where the user is writing. That one is not a huge deal to leave unimplemented, I suppose – the caret will just stay where it is using the previous implementation. The bigger ones are `firstRectForRange` and `selectionRectsForRange` which UIKit needs to do pretty much everything except inserting text at the caret: selecting, deleting, inserting/removing spaces

Reply:
Okay, so you are right, text long enough to scroll does cause some issues. I pushed up some changes to deal with this:

1. Offset the selection rects by the current scroll offset
2. Cache the selection rects so that they are only recalculated if needed
3. Wait for the scroll direction to be `.idle` so that scrolling is smooth